### ISSUE TITLE
[ integration ] expose python api

### ISSUE THREAD

**Issue by vsoch** [asciinema/asciinema#300]
This is probably a no brainer, but I'd like to be able to (programatically from within Python) have an easy way to control asciinema recording, so I can integrate it into my client's functions without needing to send commands to subprocess. I started to trace what was happening and come up with something (rather hacky):

```python
    import asciinema.config as aconfig
    from asciinema.api import Api
    from asciinema.commands.record import RecordCommand

    cfg = aconfig.load()
    api = Api(cfg.api_url, os.environ.get("USER"), cfg.api_token)

    if not filename:
        filename = generate_temporary_file(prefix='asciinema')

    # Create dummy class to pass in as args

    class settings(object):
        filename = filename
        env = os.environ
        rec_stdin = False
        command = os.environ['SHELL']
        title = title or "Helpme Recording"
        assume_yes = False 
        quiet = False
        max_wait = 10

    args = settings()
    return_code = RecordCommand(api, args=args)
```
but stopped because I was worried about not completely mimicking the parser that is defined [here](https://github.com/asciinema/asciinema/blob/develop/asciinema/__main__.py#L91). If we could have discussion about how this might be done, I have a few ideas!

*Expose Parser*
 you could expose functions to just return the parser for each command, that way a function like mine could do something like:

```python
parser = get_record_parser()
args = parser.parse_args(["arg1", "--val1","flag1"])
```
*Python API?*
As a developer, it would be really awesome to see documentation and examples for how to control from within Python. Given that we call a subshell, I think it should technically work to record from /bin/bash (or `$SHELL`) and then do it's thing and exit right back where it left off :)

*Expose Generation Function*
It would be more intuitive to expose a generation function that either prepares a dummy parser (akin to what I was trying to do) to pass on to RecordCommand, or a customization of RecordCommand so it takes more nicely named arguments, e.g.,

```python
class RecordCommand

    def __init__(self, api, filename, title="HelpMe Recording", quiet=True):
        Command.__init__(self, quiet)
        self.api = api
        self.filename = filename
       ...
```
etc.! What I'm probably going to try is overriding the __init__ function before I call it with something like this, and tweak until I have the minimum set of variables needed to get it working, haha.

And thank you so much for asciinema! It continues to be one of my [favorite things](https://asciinema.org/~vs), and I'm finally having some time to integrate it into some cool tools :)

**Comment by vsoch** [comment]
hey @sickill ! Yes this would be a really great feature to have. If you are interested, I got (most of it) working here, just for the commands that I needed (record and upload) for this library --> https://vsoch.github.io/helpme/helper-github specifically for [record](https://github.com/vsoch/helpme/blob/master/helpme/action/record.py#L80) and [submit](https://github.com/vsoch/helpme/blob/master/helpme/action/submit.py#L23). I'm really excited about using asciinema in this way - this client prompts the user for asking a question, records the terminal if they want to show an issue, and then submits to uservoice or Github. I can do other clients if anyone has ideas. Could you tell me more about the API you are developing for writing the files? What are example use cases of when we would want to do this?

**Comment by sickill** [comment]
Hey I just released 2.0.2 with this high level API for recording: https://github.com/asciinema/asciinema/blob/develop/asciinema/__init__.py#L19

It's used by RecordCommand: https://github.com/asciinema/asciinema/blob/develop/asciinema/commands/record.py#L69-L78

Most of the arguments are optional, they have reasonable defaults, the only required one is path for the resulting asciicast file.
